pages/portfolio.js

```diff
@@ -0,0 +1,155 @@
+/* global BACKEND_URL, fetch */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Layout from '../components/layout/main';
+import Background from '../components/content/background';
+
+import Works from '../components/portfolio/works';
+
+import { works } from '../main.config';
+
+const transformateCategories = (choosenCategory, existCategories) => {
+  const categories = [];
+
+  existCategories.forEach((existCategory) => {
+    choosenCategory.forEach((category) => {
+      if (category.toLowerCase() === existCategory.toLowerCase() && categories.indexOf(existCategory) === -1) {
+        categories.push(existCategory);
+      }
+    });
+  });
+
+  return categories.length ? categories : existCategories;
+}
+
+export default class Portfolio extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      ...this.getFilterList.call(this)
+    };
+
+    this.filterWorks = this.filterWorks.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.activeAllWorks = this.activeAllWorks.bind(this);
+    this.disabledBtn = this.disabledBtn.bind(this);
+  }
+
+  componentDidMount() {
+    const subnavigation = document.querySelector('.navigation-item.current > .subnavigation');
+    subnavigation.style.display = 'none';
+    subnavigation.parentElement.classList.add('is-link');
+  }
+
+  getFilterList() {
+    const { url } = this.props;
+    const filterList = [];
+    const choosenCategory = url.query.choosen;
+
+    works.forEach((work) => {
+      work.category.main.forEach((category) => {
+        if (filterList.indexOf(category) === -1) {
+          filterList.push(category);
+        }
+      });
+    });
+
+    const activeWorks = choosenCategory && choosenCategory !== 'All'
+      ? transformateCategories(choosenCategory.split(','), filterList)
+      : filterList;
+
+    return { activeWorks, filterList };
+  }
+
+  filterWorks(work) {
```

Comment:
@dmytropanontko , the name of the function is confusing. Let's call it `worksCountFor`.
Shouldn't it accept category instead of work?